inc/authldap.class.php

```diff
@@ -1654,19 +1654,50 @@ static function searchForUsers($ds, $values, $filter, $attrs, &$limitexceeded, &
       $count    = 0;  //Store the number of results ldap_search
 
       do {
+         $filter = Toolbox::unclean_cross_side_scripting_deep(Toolbox::stripslashes_deep($filter));
          if (self::isLdapPageSizeAvailable($config_ldap)) {
-            ldap_control_paged_result($ds, $config_ldap->fields['pagesize'], true, $cookie);
+            if (version_compare(PHP_VERSION, '7.3') < 0) {
+               //prior to PHP 7.3, use ldap_control_paged_result
+               ldap_control_paged_result($ds, $config_ldap->fields['pagesize'], true, $cookie);
+               $sr = @ldap_search($ds, $values['basedn'], $filter, $attrs);
+            } else {
+               //since PHP 7.3, send serverctrls to ldap_search
+               $controls = [
+                  [
+                     'oid'       =>LDAP_CONTROL_PAGEDRESULTS,
+                     'isritical' => true,
+                     'value'     => [
+                        'size'   => $config_ldap->fields['pagesize'],
+                        'cookie' => $cookie
+                     ]
+                  ]
+               ];
+               $sr = @ldap_search($ds, $values['basedn'], $filter, $attrs, 0, -1, -1, LDAP_DEREF_NEVER, $controls);
```

Comment:
Accordign to PHPDoc, should be.
```suggestion
               $sr = @ldap_search($ds, $values['basedn'], $filter, $attrs, 0, 0, 0, LDAP_DEREF_NEVER, $controls);
```

Reply:
Which doc? I've used default from https://www.php.net/manual/en/function.ldap-search.php (since nothing was passed before)

Reply:
Indeed. I read parameters description of this same page and it says that 0 could be use for unlimited results. I guess 0 and -1 acts the same.